LogMonitor/src/LogMonitor/JsonProcessor.cpp

```diff
@@ -0,0 +1,284 @@
+
+    //
+    // Fill the new EventLog source object, with its attributes
+    //
+    if (!SourceEventLog::Unwrap(Attributes, *sourceEventLog))
+    {
+        logWriter.TraceError(
+            L"Error parsing configuration file. "
+            L"Invalid EventLog source (it must have a non-empty 'channels')"
+        );
+    }
+
+    Sources.push_back(std::reinterpret_pointer_cast<LogSource>(std::move(sourceEventLog)));
+}
+
+// Function to handle File type
+void handleFile(
+    _In_ const boost::json::value& source,
+    _In_ AttributesMap& Attributes,
+    _Inout_ std::vector<std::shared_ptr<LogSource>>& Sources
+) {
+    std::string directory = source.as_object().at("directory").as_string().c_str();
+    std::string filter = source.as_object().at("filter").as_string().c_str();
+    bool includeSubdirs = source.at("includeSubdirectories").as_bool();
+    std::string customLogFormat = source.as_object().at("customLogFormat").as_string().c_str();
+
+    Attributes[JSON_TAG_DIRECTORY] = reinterpret_cast<void*>(new std::wstring(Utility::string_to_wstring(directory)));
+    Attributes[JSON_TAG_FILTER] = reinterpret_cast<void*>(new std::wstring(Utility::string_to_wstring(filter)));
+    Attributes[JSON_TAG_INCLUDE_SUBDIRECTORIES] = reinterpret_cast<void*>(
+        new std::wstring(includeSubdirs ? L"true" : L"false")
+        );
+    Attributes[JSON_TAG_CUSTOM_LOG_FORMAT] = reinterpret_cast<void*>(
+        new std::wstring(
+            Utility::string_to_wstring(customLogFormat)
+        )
+        );
+
+    std::shared_ptr<SourceFile> sourceFile = std::make_shared< SourceFile>();
+    //
+    // Fill the new FileLog source object, with its attributes
+    //
+    if (!SourceFile::Unwrap(Attributes, *sourceFile))
+    {
+        logWriter.TraceError(
+            L"Error parsing configuration file. "
+            L"Invalid File source (it must have a non-empty 'channels')"
+        );
+    }
+
+    Sources.push_back(std::reinterpret_pointer_cast<LogSource>(std::move(sourceFile)));
+}
+
+// Function to handle ETW type
+void handleETW(
+    _In_ const boost::json::value& source,
+    _In_ AttributesMap& Attributes,
+    _Inout_ std::vector<std::shared_ptr<LogSource>>& Sources
+) {
+    bool multiLine = source.at("eventFormatMultiLine").as_bool();
+    std::string customLogFormat = source.at("customLogFormat").as_string().c_str();
+
+    Attributes[JSON_TAG_FORMAT_MULTILINE] = reinterpret_cast<void*>(new std::wstring(multiLine ? L"true" : L"false"));
+    Attributes[JSON_TAG_CUSTOM_LOG_FORMAT] = reinterpret_cast<void*>(
+        new std::wstring(
+            Utility::string_to_wstring(customLogFormat)
+        )
+        );
+
+    std::vector<ETWProvider>* etwProviders = new std::vector<ETWProvider>();
+
+    const boost::json::array& providers = source.at("providers").as_array();
+    for (const auto& provider : providers) {
+        std::string providerName = provider.as_object().at("providerName").as_string().c_str();
```

Comment:
qq: btw, what happens when someone mixes the cases the names eg. `ProviderName` instead of `providerName`?